scripts/create_droplist.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Takes a list of urls from a file and transforms them using one of the
+predefined transformations, writing the results out to another txt file.
+"""
+
+from argparse import ArgumentParser
+import logging
+from pathlib import Path
+from urllib import parse
+
+
+def parse_arguments():
+    parser = ArgumentParser(description=__doc__)
+    parser.add_argument('--input', '-i', type=Path, required=True,
+                        help='the input file.')
+    parser.add_argument('--output', '-o', type=Path, required=True,
+                        help='the output file.')
+    parser.add_argument('--transformation', '-t', type=str, required=True,
+                        help='the transformation pattern to apply')
+    parser.add_argument('--log-level', '-L', type=str, default='info',
+                        choices=['debug', 'info', 'warning',
+                                 'error', 'critical'],
+                        help='the logging level.')
+    args = parser.parse_args()
+    if not args.input.is_file():
+        parser.error('The input file must exist.')
+    return args
+
+
+def nepszava_transformation(input_url: str):
+    """
+    Transformation example:
+    input:
+    https://nepszava.hu/json/cikk.json?id=1001322_elindult-a-bekemenet
+    output:
+    http://nepszava.hu/1001322_elindult-a-bekemenet

```

Comment:
I was wondering: is this systematic in the sense that we don't get the json-type URLs in our data? Would it make sense to keep both the input and the output?

Reply:
Hmmm...

In nepszava we have the following patterns:

* the one seen in the output
* same, just with https
* the same but with cikk/ inserted to between the domain and the path.

These three are duplicates of each other, but not every article exists in all three forms (some do).

I just realized that if we want to keep the documents from webaratás, then we should filter out all three variants. :(

A fourth pattern is:
http://nepszava.hu/articles/article.php?id=238077

These have 6 digit numbers, while the rest have 7 digit numbers, so they don't seem to match.

The common crawl has no urls with nepszava.hu/json